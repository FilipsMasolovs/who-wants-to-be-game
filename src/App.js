import React from 'react'
import Spotlight from './assets/components/spotlight'
import Copyright from './assets/components/copyright'
import About from './assets/components/about'
import Login from './assets/components/login'
import Game from './assets/components/game'
import Fail from './assets/components/game/fail'
import Finished from './assets/components/game/finished'

class App extends React.Component {
  constructor (props) {
    super(props)
    this.state = {
      isAboutVisible: false,
      isGameStarted: false,
      isGameFailed: false,
      isGameFinished: false,
      playerName: '',
      placeholder: '',
      points: 0
    }
    this.openAbout = this.openAbout.bind(this)
    this.startGame = this.startGame.bind(this)
    this.handleGameFail = this.handleGameFail.bind(this)
    this.restartGame = this.restartGame.bind(this)
    this.handleGameFinish = this.handleGameFinish.bind(this)
  }

  openAbout (e) {
    if (e.target !== e.currentTarget) {
     return
    }
    this.setState(prevState => ({
      isAboutVisible: !prevState.isAboutVisible
    }))
  }

  startGame (playerName) {
    if (!playerName) {
      this.setState({
        placeholder: 'Please enter your name!'
      })
      return
    }
    this.setState({
      isGameStarted: true,
      playerName: playerName
    })
  }

  handleGameFail () {
    this.setState({
      isGameStarted: false,
      isGameFailed: true,
      playerName: '',
      points: 0
    })
  }

  restartGame (e) {
    if (e.target !== e.currentTarget) {
      return
    }

    this.setState({
      isGameStarted: false,
      isGameFailed: false,
      isGameFinished: false,
      playerName: '',
      points: 0
    })
  }

  handleGameFinish (e) {
    this.setState({
      isGameStarted: false,
      isGameFinished: true,
      points: e,
      placeholder: ''
    })
  }

  render () {
    return (
      <div className='wwvce-container'>
        {!this.state.isGameStarted && <Spotlight />}
        {!this.state.isGameStarted && <Login onOpenAbout={this.openAbout} onStartGame={this.startGame} placeholder={this.state.placeholder} />}
        {this.state.isGameStarted && <Game onGameFail={this.handleGameFail} onGameFinish={this.handleGameFinish} />}
        {this.state.isAboutVisible && <About onCloseAbout={this.openAbout} />}
        {this.state.isGameFailed && <Fail onRestartGame={this.restartGame} />}
        {this.state.isGameFinished && <Finished playerName={this.state.playerName} points={this.state.points} onRestartGame={this.restartGame} />}
        <Copyright />
      </div>
    )
  }
}

export default App