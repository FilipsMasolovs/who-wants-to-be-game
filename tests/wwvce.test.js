import React from 'react'
import renderer from 'react-test-renderer'
import App from '../src/App'

describe('Tests game component rendering.', () => {
  test('App component render.', () => {
    const tree = renderer
      .create(<App />)
      .toJSON()
      expect(tree).toMatchSnapshot()
    })
})
