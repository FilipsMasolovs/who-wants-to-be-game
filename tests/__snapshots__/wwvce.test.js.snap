// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests game component rendering. App component render. 1`] = `
<div
  className="wwvce-container"
>
  <div
    className="wwvce-spotlight"
  >
    <img
      alt="Who Wants to Be a Visual Composer expert?"
      className="wwvce-spotlight-logo"
      src=""
    />
    <h1
      className="wwvce-spotlight-title"
    >
      Who Wants to Be a Visual Composer expert?
    </h1>
  </div>
  <div
    className="wwvce-login"
  >
    <div
      className="wwvce-login-inner"
    >
      <span
        className="wwvce-login-title"
      >
        Please enter your name:
      </span>
      <input
        className="wwvce-login-input"
        onChange={[Function]}
        onKeyPress={[Function]}
        placeholder=""
        type="text"
      />
    </div>
    <div
      className="wwvce-login-inner wwvce-login-inner-controls"
    >
      <button
        className="wwvce-login-start-game-button"
        onClick={[Function]}
      >
        START
      </button>
      <button
        className="wwvce-login-open-about-button"
        onClick={[Function]}
      >
        Learn more...
      </button>
    </div>
  </div>
  <a
    className="wwvce-copyright"
    href="https://mr.masolov.com/"
    target="blank"
  >
    by: MrMasolov
  </a>
</div>
`;
